hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/navBar/navBar.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { Layout, Menu, Spin } from 'antd';
+import {
+  BarChartOutlined,
+  ClusterOutlined,
+  ContainerOutlined,
+  DashboardOutlined,
+  DatabaseOutlined,
+  DeploymentUnitOutlined,
+  FolderOpenOutlined,
+  InboxOutlined,
+  LayoutOutlined,
+  PieChartOutlined
+} from '@ant-design/icons';
+import { useLocation, Link } from 'react-router-dom';
+
+
+import logo from '@/logo.png';
+import { showDataFetchError } from '@/utils/common';
+import { AxiosGetHelper, cancelRequests } from '@/utils/axiosRequestHelper';
+
+import './navBar.less';
+
+
+// ------------- Types -------------- //
+type NavBarProps = {
+  collapsed: boolean;
+  onCollapse: (arg0: boolean) => void;
+}
+
+const NavBar: React.FC<NavBarProps> = ({
+  collapsed = false,
+  onCollapse = () => { }
+}) => {
+  const [isHeatmapEnabled, setIsHeatmapEnabled] = useState<boolean>(false);
+  let cancelDisabledFeatureSignal: AbortController;
```

Comment:
The cancelDisabledFeatureSignal is declared with let but is never reassigned, potentially leading to unexpected behavior if the component re-renders.

```suggestion
  const [cancelDisabledFeatureSignal, setCancelDisabledFeatureSignal] = useState<AbortController>(new AbortController());

```

Reply:
We would not need to use state in this scenario. As discussed in https://github.com/apache/ozone/pull/7088#discussion_r1732546828, this fetch method handler will only be called in the initial component mount.
During that time this variable will get set with the current request signal.